nix-build -A swift       BUILD FAILS

## Issue description

/tmp/nix-build-swift-4.0.3.drv-0/src/lldb/include/lldb/Utility/TaskPool.h:122:12: error: no member named 'bind' in namespace 'std'

### Steps to reproduce

git clone git@github.com:NixOS/nixpkgs.git
cd nixpkgs
nix-build -A swift

## Technical details

The hydra build has the same problem I had, which you can see at the end of the build log:

Build log: https://hydra.nixos.org/build/69887235/nixlog/1

Last successful build: https://hydra.nixos.org/build/69575672

--- a/pkgs/development/compilers/swift/default.nix
+++ b/pkgs/development/compilers/swift/default.nix
@@ -145,7 +145,6 @@ let
     url = "https://github.com/apple/swift-llbuild/commit/303a89bc6da606c115560921a452686aa0655f5e.diff";
     sha256 = "04sw7ym1grzggj1v3xrzr2ljxz8rf9rnn9n5fg1xjbwlrdagkc7m";
   };
-
 in
 stdenv.mkDerivation rec {
   name = "swift-${version_friendly}";
@@ -246,6 +245,9 @@ stdenv.mkDerivation rec {
 
     # https://bugs.swift.org/browse/SR-5779
     sed -i -e 's|"-latomic"|"-Wl,-rpath,${clang.cc.gcc.lib}/lib" "-L${clang.cc.gcc.lib}/lib" "-latomic"|' swift/cmake/modules/AddSwift.cmake
+
+    # https://bugs.swift.org/browse/SR-4838
+    sed -i -e '30i#include <functional>' lldb/include/lldb/Utility/TaskPool.h
 
     substituteInPlace clang/lib/Driver/ToolChains.cpp \
       --replace '  addPathIfExists(D, SysRoot + "/usr/lib", Paths);' \

--- a/pkgs/development/compilers/swift/default.nix
+++ b/pkgs/development/compilers/swift/default.nix
@@ -289,7 +289,7 @@ stdenv.mkDerivation rec {
 
     # TODO: Use wrappers to get these on the PATH for swift tools, instead
     ln -s ${clang}/bin/* $out/bin/
-    ln -s ${targetPackages.stdenv.cc.bintools}/bin/ar $out/bin/ar
+    ln -s ${targetPackages.stdenv.cc.bintools.bintools_bin}/bin/ar $out/bin/ar
 
     wrapProgram $out/bin/swift \
       --suffix C_INCLUDE_PATH : $out/lib/swift/clang/include \

